Sprint-3/todo-list/script.js

```diff
@@ -1,25 +1,72 @@
-function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
-}
+const removeAllButton = document.getElementById("remove-all-completed");
+const form = document.getElementById("todo-form");
+const input = document.querySelector(".input");
+const todoList = document.getElementById("todo-list");
 
-// These are the same todos that currently display in the HTML
-// You will want to remove the ones in the current HTML after you have created them using JavaScript
 let todos = [
   { task: "Wash the dishes", completed: false },
   { task: "Do the shopping", completed: false },
 ];
 
-populateTodoList(todos);
+// Populate the todo list with initial tasks
+function populateTodoList(todos) {
+  todos.forEach((item) => {
+    addTodoToDOM(item.task, item.completed);
+  });
+}
+
+// Add todo item to the DOM
+function addTodoToDOM(task, completed = false) {
+  const newTodo = document.createElement("li");
+  newTodo.classList.add("todo-item");
+  if (completed) newTodo.classList.add("done");
+  newTodo.innerHTML = `
+    <span class="badge bg-primary rounded-pill">${task}</span>
+    <button class="status">${completed ? "UNDO" : "DONE"}</button>
+    <button class="delete">DELETE</button>
+  `;
+  todoList.appendChild(newTodo);
+}
 
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
+// Add new todo from the input field
 function addNewTodo(event) {
-  // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+
+  const taskValue = input.value.trim();
+  if (taskValue) {
```

Comment:
Often if someone presses a button, they expect it to do something, and if it quietly doesn't, they think things are broken.

If they haven't typed _any_ text into the text box, and press the button, they maybe don't expect anything to happen. But if they typed a few spaces into the box, and pressed the button, they maybe expected a new to do item to appear.

Your code just doesn't do anything if they put in spaces and press the button.

Maybe this is fine, or maybe it would be useful to indicate to them why you didn't do anything when they pressed the button.

What do you think? How could you maybe tell them something is wrong? Or maybe it's fine to just do nothing - but something to think about.

Reply:
Thank you! I added alert for empty text.